numba/tests/test_dummyarray.py

```diff
@@ -208,6 +208,51 @@ def test_reshape_3d1d(self):
         self.assertEqual(got.shape, expect.shape)
         self.assertEqual(got.strides, expect.strides)
 
+    def test_reshape_infer2d2d(self):
+        nparr = np.empty((4, 5))
+        arr = Array.from_desc(0, nparr.shape, nparr.strides,
+                              nparr.dtype.itemsize)
+        expect = nparr.reshape(5, 4)
```

Comment:
Although this is equal to the shape we'd expect, it does feel a little like we're not quite testing like-for-like here, because we're not creating the expected array with a missing dimension. Perhaps:
```suggestion
        expect = nparr.reshape(-1, 4)
```
would be more explicit that we're testing like-for-like?

(similar applies to the subsequent four `nparr.reshape` calls below too).

Reply:
I had been thinking that by explicitly putting the result of the inferred dimension in the expected, it would make it clear that we're ensuring that the logic for working out the missing dimension is working appropriately. But I'm realizing now it's sufficient to match the numpy behavior. I agree that is more explicit in testing like-for-like!